src/Illuminate/Database/Connectors/SqlServerConnector.php

```diff
@@ -166,10 +166,10 @@ protected function buildConnectString($driver, array $arguments)
      */
     protected function buildHostString(array $config, $separator)
     {
-        if (isset($config['port']) && ! empty($config['port'])) {
-            return $config['host'].$separator.$config['port'];
-        } else {
+        if (empty($config['port'])) {
             return $config['host'];
+        } else {
```

Comment:
`else` statement could be removed because of early return:

```php
if (empty($config['port'])) {
    return $config['host'];
}

return $config['host'].$separator.$config['port'];
```